src/Select/Select.tsx

```diff
@@ -26,20 +28,20 @@ export interface SelectProps extends BlockProps {
   children: any
 }
 
-export const injectComponent = (component: React.ReactElement<any>, props: InjectValueAndHandler) => {
-  return React.cloneElement(component, {
+export const injectComponent = (component: ReactElement<any>, props: InjectValueAndHandler) => {
+  return cloneElement(component, {
     ...props,
     ...component.props
```

Comment:
What's the reason for spreading in the original component props here? cloneElement has this behaviour out of the box, only difference from the logic here is the in cloneElement, original component props will be overwritten by merged in props.